lms/services/course.py

```diff
@@ -140,19 +140,33 @@ def search(  # noqa: PLR0913, PLR0917
             h_userid=h_userid,
         ).all()
 
-    def get_organization_courses(
-        self, organization: Organization, h_userid: str | None
+    def get_courses(
+        self,
+        h_userid: str | None,
+        organization: Organization | None = None,
     ):
+        """Get a list of unique courses.
+
+        :param organization: organization the courses belong to.
+        :param h_userid: only courses this user has access to.
+        """
         courses_query = self._search_query(
-            organization_ids=[organization.id],
+            organization_ids=[organization.id] if organization else None,
             h_userid=h_userid,
             limit=None,
         )
-        return (
-            # Deduplicate courses by authority_provided_id, take the last updated one
+
+        # Deduplicate courses by authority_provided_id, take the last updated one
+        deduplicated_courses = select(Course.id).select_from(
             courses_query.distinct(Course.authority_provided_id)
             .order_by(Course.authority_provided_id, Course.updated.desc())
-            .all()
+            .subquery()
+        )
+
+        return (
+            self._db.query(Course)
+            .filter(Course.id.in_(deduplicated_courses))
+            .order_by(Course.lms_name, Course.id)
```

Comment:
Order by the column we'll paginate with. Maybe in the future this could be an arbitrary set of columns.

Let's not try to build an arbitrary pagination _system_ with the first paginated endpoint.

Reply:
We need to columns to make for a more stable order than just the name. This could eventually become:

 Course.name, Course.**term**, Course.id but I'd always include the ID there for good measure at the end.

Reply:
> We need to columns to make for a more stable order than just the name. 

You mean _two_ columns, right?

Reply:
Yes :facepalm: